src/fromager/commands/lint_requirements.py

```diff
@@ -0,0 +1,52 @@
+import fileinput
+import glob
+import itertools
+import logging
+import sys
+
+import click
+from packaging.requirements import InvalidRequirement, Requirement
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument("constraints_wildcard", type=click.Path(exists=False))
+@click.argument("requirements_wildcard", type=click.Path(exists=False))
+def lint_requirements(constraints_wildcard: str, requirements_wildcard: str) -> None:
+    """
+    Command to lint the constraints.txt and requirements.txt files
+    This command takes two wildcard paths for constraints.txt and requirements.txt respectively.
+    It checks the formatting of these files and reports issues if found.
+    """
+    # Exit if user does not provide wildcard paths
+    if not constraints_wildcard or not requirements_wildcard:
+        logger.error("path for requirements.txt or constraints.txt is missing")
+        sys.exit(1)
+
+    # Get all the files to check in a list
+    files_to_check = list(
+        itertools.chain.from_iterable(
+            glob.glob(f) for f in [constraints_wildcard, requirements_wildcard]
+        )
+    )
+
+    if len(files_to_check) == 0:
+        logger.error("no constraints.txt or requirements.txt found in given paths")
+        sys.exit(1)
+
+    flag = True
+
+    for line in fileinput.input(files_to_check):
```

Comment:
We had a copy of the implementation in the downstream linter, but now that we're moving it here we could use the parsing logic in https://github.com/python-wheel-build/fromager/blob/main/src/fromager/requirements_file.py instead.

Reply:
Ah didn't notice that!
I have made changes in my latest code push